File: DSA 4th Year/TUF/Patterns/star_diamond.cpp
#include <bits/stdc++.h>
using namespace std;
int main(){

    int n=3;
    for (int  i = 1; i <= n; i++)
    {
        int spaces=n-i;
        for(int s=1;s<=n-i;s++){
            cout<<" ";
        }
        for (int  j = 1; j <= 2*i-1; j++)
        {
            cout<<"*";
        }
        cout<<endl;
        
    }

    for (int  i = 1; i <= n; i++)
    {
        int spaces=i-1;
        while(spaces!=0){
            cout<<" ";
            spaces--;
        }
        for (int  j = 1; j <= (2*n)-(2*i)+1; j++)
        {
            cout<<"*";
        }
        cout<<endl;
        
    }
    
    return 0;

}

File: DSA 4th Year/TUF/Patterns/star_diamond.py
n=3
for i in range(1,n+1):
    # spaces print 
    for spaces in range (1,n-i+1):
        print(" ",end="")
    for j in range(1,2*i):
        print("*",end="")

    print()
for i in range(1,n+1):
    # spaces print 
    for spaces in range (1,i):
        print(" ",end="")
    for j in range(0,2*n-(2*i)+1):
        print("*",end="")

    print()